core/src/banking_stage.rs

```diff
@@ -1212,8 +1220,11 @@ mod tests {
                     Blocktree::open(&ledger_path)
                         .expect("Expected to be able to open database ledger"),
                 );
+                let mut poh_config = PohConfig::default();
+                // limit the tick to 1 to prevent clearing working_bank at PohRecord then PohRecorderError(MaxHeightReached) at BankingStage
+                poh_config.target_tick_count = Some(1);
```

Comment:
Good catch, quite a subtle bug, especially for your first time through the code base :)

Instead of the magic number 1, can we use something like `genesis_block.ticks_per_slot - 1`, to make it clear that there's nothing special about this number, it just needs to be < genesis_block.ticks_per_slot

Reply:
sure thing! done at https://github.com/solana-labs/solana/pull/6251/commits/a9138d1efdcaff8d1b8979c638c13aefab0ed444

Reply:
Well, I couldn't directly use genesis_block.ticks_per_slot because subtracting -1 from it still causes the race condition, so I chose different one: b07bf58e0cfc580f65e08ee731ce88fc2af8669e

Reply:
@ryoqun, ah yes, it's because of a special case where slot 0 generates one less tick than all the other slots. For instance, if `genesis_block.ticks_per_slot` is equal to 10, then the first slot will have `bank.max_tick_height` of `9`, so PoH will, starting at 0, generate a tick for tick_heights `0->1, 1->2,  2->3...8->9`, which is only 9 ticks. The next slot, slot 1, will then generate 10 ticks because it's max tick height is 19, and it will start generating a tick for tick heights `9->10, 10->11, ... 18 -> 19`. This is all because in the past we wanted to keep the first tick in a slot as 0-indexed. 

This open PR here actually will address this issue: https://github.com/solana-labs/solana/pull/6263.

Reply:
Thanks for very detailed explanation! I'll look at it. :D